src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -0,0 +1,408 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static class OpenSsl
+    {
+        #region structures
+        [StructLayout(LayoutKind.Sequential)]
+        private struct SslContext
+        {         
+            internal IntPtr sslPtr;
+            internal IntPtr readBioPtr;
+            internal IntPtr writeBioPtr;
+            internal bool isServer;          
+        }
+        #endregion
```

Comment:
I'll reiterate the comments I made in the other PR: in summary, I don't understand why we're using a struct for this.  We're allocating native memory to store it, and then any time we need the data, we're having to marshal the data back to managed.  We should instead just use a class, using a GCHandle to keep it alive and have a pointer we can pass around in native.  This will also help with ensuring we don't accidentally double-free the pointers contained, as we can null them out after we've free'd them.